addons/website_slides/static/src/js/public/components/slide_quiz_finish_dialog/slide_quiz_finish_dialog.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates xml:space="preserve">
+
+    <t t-name="website_slides.SlideQuizFinsihDialog" owl="1">
+        <Dialog 
+            bodyClass="'d-flex p-0'" 
+            contentClass="'shadow-lg'"
+            footer="false" 
+            header="false" 
+            size="'md o_wslides_quiz_modal pt-5'" 
+            technical="false" 
+        >
+            <a role="button" class="o_wslides_quiz_modal_close_btn btn-close position-absolute" aria-label="Close" t-on-click.prevent="() => this.props.close()"/>
+            <div class="o_wslides_quiz_success_image d-none d-md-flex flex-shrink-0">
+                <img class="o_wslides_quiz_modal_hero" src="/website_slides/static/src/img/quiz_modal_success.svg" alt=""/>
+            </div>
+            <div class="d-flex flex-column flex-grow-1 justify-content-between ps-md-5 p-3 overflow-auto">
+                <div>
+                    <h1 class="o_wslides_quiz_modal_title mt-3 display-4 fw-bold" t-out="state.title"/>
+                    <div class="pb-3">
+                        <h4 class="o_wslides_quiz_modal_xp_gained pb-2 d-flex fw-bold fade" t-att-class="{'show': state.animateKarmaGain, 'in': state.animateKarmaGain}">
+                            <t t-if="props.quiz.quizKarmaWon > 0">
+                                You gained <span class="badge rounded-pill text-bg-success text-white fw-bold ms-2 me-1"><t t-out="props.quiz.quizKarmaWon"/> XP</span>!
+                            </t>
+                            <t t-else="">You did it!</t>
+                        </h4>
+                        <div class="mt-5 mb-4">
+                            <div class="progress position-relative">
+                                <div class="tooltip fade show bs-tooltip-top" role="tooltip" t-attf-style="position: absolute; left: #{state.tooltipLeft}%; transform: translate(-50%, -100%); margin: 0px; ">
+                                    <div class="tooltip-arrow" style="position: absolute; left: 50%; transform: translate(-50%, 0px);"/>
+                                    <div class="tooltip-inner" t-out="state.tooltipValue"/>
+                                </div>
+                                <div t-att-class="{'progress-bar': true, 'no-transition': state.progressBarTransition}" role="progressbar" t-att-aria-valuenow="props.quiz.rankProgress.previous_rank.progress" aria-valuemin="0" aria-valuemax="100"
+                                    t-attf-style="width: #{state.progressBarWidth}%"/>
+                            </div>
```

Comment:
I would extract these in a separate component, about half the state and code of the dialog is actually for this progressbar alone. 

With this base, a few things get simplified:
```js
export class SlideXPProgressBar extends Component {
    static props = {
        rankProgress: Object,
    }
    static template = "website_slides.SlideXPProgressBar";

    setup() {
        super.setup();
        this.state = useState({
            progressBarWidth: 0,
            progressBarTransition: false,
            tooltipValue: this.props.rankProgress.previous_rank.karma,
            tooltipLeft: 0,
        });
        onMounted(() => { this._animateProgressBar(); });
    }
```

Calling it is not complex either
```suggestion
                            <SlideXPProgressBar rankProgress="props.quiz.rankProgress"/>
```